File: monitoring_examples/src/simple_monitor.cpp
#include <ros/ros.h>
#include <monitoring_core/monitor.h>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "simple_monitor");
  ros::NodeHandle nh;

  Monitor monitor(nh, "simple_monitor_example");

  ros::Rate loop_rate(10);
  int c = 0;
  while (ros::ok()){
    monitor.addValue("count", c, "SIUnits", 0.0);
    c++;
    monitor.publish();
    loop_rate.sleep();
  }


  ROS_INFO("Hello world!");
}

File: monitoring_examples/scripts/simple_monitor.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import rospy
from monitoring_core.monitor import Monitor
from std_msgs.msg import String


def talker():
    rospy.init_node('simple_monitor', anonymous=True)
    rate = rospy.Rate(10) # 10hz
    monitor = Monitor("simple_monitor_python")
    c = 0
    while not rospy.is_shutdown():
        monitor.addValue("count", c, "SIUnits", 0.0)
        c += 1
        rospy.loginfo("c: %i", c)
        monitor.publish()
        rate.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException:
        pass
